tests/benchmark/test_worst_bytecode.py

```diff
@@ -315,3 +318,188 @@ def test_worst_initcode_jumpdest_analysis(
+        sender=pre.fund_eoa(),
+    )
+
+    state_test(
+        env=env,
+        pre=pre,
+        post={},
+        tx=tx,
+    )
+
+
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.parametrize(
+    "opcode",
+    [
+        Op.CREATE,
+        Op.CREATE2,
+    ],
+)
+def test_worst_creates_collisions(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    fork: Fork,
+    opcode: Op,
+):
+    """Test the CREATE and CREATE2 collisions performance."""
+    env = Environment()
+
+    # We deploy a "proxy contract" which is the contract that will be called in a loop
+    # using all the gas in the block. This "proxy contract" is the one executing CREATE2
+    # failing with a collision.
+    # The reason why we need a "proxy contract" is that CREATE(2) failing with a collision will
+    # consume all the available gas. If we try to execute the CREATE(2) directly without being
+    # wrapped **and capped in gas** in a previous CALL, we would run out of gas very fast!
+    init_code = Op.PUSH0 + Op.PUSH0 + Op.RETURN
+    init_code_size = len(init_code)
+    padded_init_code = init_code + b"\x00" * (32 - init_code_size)
+    # The proxy contract basically does a quick MSTORE to configure the init code and
+    # calls CREATE(2) with such init code. The current call frame gas will be exhausted because of
+    # the collision. For this reason the caller will carefully give us the minimal gas necessary
+    # to execute the CREATE(2) and not waste any extra gas in the CREATE(2)-failure.
+    #
+    # Note that these CREATE(2) calls will fail because in (**) below we pre-alloc contracts
+    # with the same address as the ones that CREATE(2) will try to create.
+    proxy_contract = pre.deploy_contract(
+        code=Op.MSTORE(0, Bytes(padded_init_code))
+        + Op.CREATE2(value=Op.PUSH0, salt=Op.PUSH0, offset=Op.PUSH0, size=init_code_size)
+        if opcode == Op.CREATE2
+        else Op.CREATE(value=Op.PUSH0, offset=Op.PUSH0, size=init_code_size)
+    )
+
+    gas_costs = fork.gas_costs()
+    # The CALL to the proxy contract needs at a minimum the (Pectra-current) 32k of gas of the
+    # CREATE(2), plus some extra gas for satellite opcodes that we configured in that contract.
+    # The current value  was determined by inspecting the trace.
+    code_prefix = Op.PUSH3(gas_costs.G_CREATE + 29)
+    attack_block = Op.POP(
+        Op.CALL(gas=Op.DUP7, address=proxy_contract)
+    )  # DUP7 refers to the PUSH3 above.
+    code = code_loop_precompile_call(code_prefix, attack_block, fork)
+    tx_target = pre.deploy_contract(code=code)
+
+    # (**) We deploy the contract that CREATE(2) will attempt to create so any attempt will fail.
+    if opcode == Op.CREATE2:
+        addr = compute_create2_address(address=proxy_contract, salt=0, initcode=init_code)
+        pre.deploy_contract(address=addr, code=Op.INVALID)
+    else:
+        # Heuristic to have an upper bound.
+        max_contract_count = 2 * env.gas_limit // gas_costs.G_CREATE
```

Comment:
Why 2x the env limit? See comment below, we only need to deploy 1 contract I believe.